parity/upgrade.rs

```diff
@@ -18,14 +18,15 @@
 
 use semver::Version;
 use std::collections::*;
-use std::fs::File;
+use std::fs::{File, create_dir_all};
 use std::env;
 use std::io::{Read, Write};
 
 #[cfg_attr(feature="dev", allow(enum_variant_names))]
 #[derive(Debug)]
 pub enum Error {
-	CannotLockVersionFile,
+	CannotCreateConfigPath,
```

Comment:
Wrapping std::io::Error in those errors could help debugging potential problems.